helper/restore_helper.go

```diff
@@ -118,12 +127,47 @@ func (r *RestoreReader) copyAllData() (int64, error) {
 	return bytesRead, err
 }
 
+func (r *RestoreReaderImpl) getReaderType() ReaderType {
+	return r.readerType
+}
+
+func (r *RestoreReaderImpl) getFileHandle() *os.File {
+	return r.fileHandle
+}
+
+func closeAndDeletePipe(h Helper, tableOid int, batchNum int) {
+	pipe := fmt.Sprintf("%s_%d_%d", *pipeFile, tableOid, batchNum)
+	logInfo(fmt.Sprintf("Oid %d, Batch %d: Closing pipe %s", tableOid, batchNum, pipe))
+	errPipe := h.flushAndCloseRestoreWriter(pipe, tableOid)
+	if errPipe != nil {
+		logVerbose(fmt.Sprintf("Oid %d, Batch %d: Failed to flush and close pipe: %s", tableOid, batchNum, errPipe))
+	}
+
+	logVerbose(fmt.Sprintf("Oid %d, Batch %d: Attempt to delete pipe %s", tableOid, batchNum, pipe))
+	errPipe = deletePipe(pipe)
+	if errPipe != nil {
+		logError("Oid %d, Batch %d: Failed to remove pipe %s: %v", tableOid, batchNum, pipe, errPipe)
+	}
+}
+
 type oidWithBatch struct {
 	oid   int
 	batch int
 }
 
-func doRestoreAgent() error {
+type RestoreHelper interface {
```

Comment:
```suggestion
type IRestoreHelper interface {
```

Reply:
there is no convention to start interface name from I in gpbackup